### PR TITLE
Transresnet Model Release

### PR DESCRIPTION
**Pull request by klshuster**
**Patch description**
Provides the model code for Transresnet, the model introduced in [Engaging Image Captioning via Personality](https://arxiv.org/abs/1810.10665). Additionally provides access to a pretrained model in the model zoo - note this pretrained model was trained with ResNet152 images. Includes an interactive script to test out the pretrained model.

**Testing steps**
1. Ensured that this model reproduces the result from the paper (57.1 R@1 on the test set)
2. Ensured that specifying `-mf models:personality_captions/transresnet/model` correctly downloads the model from the zoo.
3. Ensured that running the interactive script generates appropriate captions for pictures.

**Logs**
```
$ python examples/eval_model.py -t personality_captions -mf models:personality_captions/transresnet/model --num-test-labels 5 -dt test

...

finished evaluating task personality_captions using datatype test
{'exs': 10000, 'accuracy': 0.5172, 'f1': 0.5998, 'hits@1': 0.517, 'hits@5': 0.824, 'hits@10': 0.908, 'hits@100': 0.998, 'bleu': 0.5057, 'hits@1/100': 0.5172, 'loss': -0.0079, 'med_rank': 1.0}
```

Note that I used internal version of dataset to really produce this log, as that is where I have access to the YFCC images.

Example of Interactive Script Output:

Screenshot 2019-05-31 12 52 26

**Other information**
Two things.

1. I encourage you to checkout this branch and try running the interactive script to make sure you can :)
2. Note that this model is unfortunately *not* based on any TorchAgent.

### CODE REVIEW

**Comment by stephenroller** [comment]
Seems like the kind of change that could use a GPU test...

**Comment by stephenroller** [comment]
Given how many images there are, you might consider following what I did in the controllable dialogue test:

https://github.com/facebookresearch/ParlAI/blob/72a17b3343f0c3b7ad8d7827d5a3d6a726ef377e/tests/nightly/gpu/test_controllable.py#L17-L18

https://github.com/facebookresearch/ParlAI/blob/72a17b3343f0c3b7ad8d7827d5a3d6a726ef377e/tests/nightly/gpu/test_controllable.py#L91


https://github.com/facebookresearch/ParlAI/blob/72a17b3343f0c3b7ad8d7827d5a3d6a726ef377e/tests/nightly/gpu/test_controllable.py#L94-L99

**Comment by klshuster** [comment]
Note: will add test, with 10 resnet images

**Comment by klshuster** [comment]
I've added a gpu test for the model with 10 images from the test set - confirmed locally that the test passes.

File: parlai/core/torch_ranker_agent.py [hunk 1]
```diff
@@ -200,9 +200,12 @@ def is_valid(self, obs):
...
     def train_step(self, batch):
         """Train on a single batch of examples."""
-        if batch.text_vec is None:
+        if batch.text_vec is None and batch.image is None:
             return
-        batchsize = batch.text_vec.size(0)
+        if batch.text_vec is not None:
+            batchsize = batch.text_vec.size(0)
+        else:
+            batchsize = len(batch.image)
```
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/core/torch_ranker_agent.py]: ahh, this is an artifact from a previous attempt at releasing this model. Though this is something that would necessarily be broken if you tried using TRA with the Personality-Captions dataset

File: tests/nightly/gpu/test_transresnet.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        self.assertEqual(
+            test['hits@5'], 0.9,
+            'test hits@5 = {}\nLOG:\n{}'.format(test['hits@5'], stdout)
+        )
+        self.assertEqual(
+            test['hits@10'], 0.9,
+            'test hits@10 = {}\nLOG:\n{}'.format(test['hits@10'], stdout)
+        )
+        self.assertGreaterEqual(
+            test['med_rank'], 1.0,
```
[the remark this answers is not in the document]
Comment (klshuster) [reply tests/nightly/gpu/test_transresnet.py]: Oops, will fix

**Comment by klshuster** [comment]
@samhumeau I've added the `no_cuda` option by default. Also, `--image-features` is no longer a relevant parameter (will discuss offline)

File: parlai/core/torch_ranker_agent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/core/torch_ranker_agent.py]: (note that i've removed these changes from this diff)

**Comment by riven314** [comment]
thanks for the great work! @klshuster 
is there any plan to release script for training/ fine-tuning the model?
